### PR TITLE
fix: force deletion of archived toggles when deleting a project

### PR DESCRIPTION
**Pull request by sighphyre**
This forces the project service to delete archived toggles when a project is deleted. Previously, when deleting a project, you were blocked if you had one or more active toggles but once they're archived the project can be deleted. Those archived toggles would persist indefinitely and when a new project was created that matched the old project name, they would immediately be bound to that project in an archived state. This was really confusing for users across teams, since they could create a project that someone else had created, used and then deleted and they would get a bunch of toggles in their project that they had never touched

### CODE REVIEW

**Review by sjaanus — APPROVED** [review]
LGTM

File: src/lib/services/project-service.ts [hunk 1]
```diff
@@ -390,6 +390,14 @@ export default class ProjectService {
             );
         }
 
+        const archivedToggles = await this.featureToggleStore.getAll({
+            project: id,
+            archived: true,
+        });
+
+        await this.featureToggleStore.batchDelete(
```
Comment (sjaanus) [inline src/lib/services/project-service.ts:398]: We should also post events for deletions

**Review by sjaanus — CHANGES_REQUESTED** [review]

File: src/lib/services/project-service.ts (hunk 1 above)
Comment (sighphyre) [reply src/lib/services/project-service.ts:398]: If we're deleting the project, where will these go?

Comment (sjaanus) [reply src/lib/services/project-service.ts:398]: Global event list.

Comment (sighphyre) [reply src/lib/services/project-service.ts:398]: Update - we have a global event log. That's where this will go

Comment (sighphyre) [reply src/lib/services/project-service.ts:398]: Good catch, I'll fix, thanks!

**Review by sjaanus — APPROVED** [review]
The code looks good, if you can get tests working.

**Pull request merged**